### PR TITLE
Add module text-classification

### PR DESCRIPTION
**Pull request by Gram21**
Done by @Lin0pan

### CODE REVIEW

**Comment by Lin0pan** [comment]
ready to review

File: text-classification/pom.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.github.ardoco.core</groupId>
+        <artifactId>parent</artifactId>
+        <version>${revision}</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (dfuchss) [inline text-classification/pom.xml:11]:
```suggestion
```

File: text-classification/pom.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            <releases>
+                <enabled>false</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+            <id>mavenSnapshot</id>
+            <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
+        </repository>
+    </repositories>
```
Comment (dfuchss) [inline text-classification/pom.xml:34]:
```suggestion
```

File: text-classification/pom.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    </parent>
+
+    <properties>
+        <maven.compiler.target>17</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+    </properties>
+
+    <groupId>org.example</groupId>
+    <version>1.0-SNAPSHOT</version>
+    <artifactId>text-classification</artifactId>
```
Comment (dfuchss) [inline text-classification/pom.xml:21]: Needs to be fixed

File: text-classification/pom.xml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            <id>mavenSnapshot</id>
+            <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.googlecode.json-simple</groupId>
+            <artifactId>json-simple</artifactId>
+            <version>1.1.1</version>
```
Comment (dfuchss) [inline text-classification/pom.xml:40]: We use jackson elsewhere (afaik)

**Review by dfuchss — CHANGES_REQUESTED** [review]
Found some things that needs to be addressed before merging

**Comment by dfuchss** [comment]
@sonatype-lift help

File: text-classification/src/main/java/docker/helper/ExecuteCmd.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+     * Run a command on the command line.
+     *
+     * @param command                  the command
+     * @param waitForResponseInSeconds the time to wait until canceling
+     * @return the result of the execution
+     */
+    @SuppressWarnings("java:S2142")
+    public static ExecuteResult runCommand(String command, int waitForResponseInSeconds) {
+        try {
+            Process process = Runtime.getRuntime().exec(command);
```
[the remark this answers is not in the document]
Comment (dfuchss) [reply text-classification/src/main/java/docker/helper/ExecuteCmd.java]: @sonatype-lift help

**Comment by dfuchss** [comment]
@Gram21 @Lin0pan maybe we have to merge `main` to the branch? The diff is not readable :)

File: text-classification/pom.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      <version>4.7.0</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-junit-jupiter</artifactId>
+      <version>4.7.0</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (dfuchss) [inline text-classification/pom.xml:44]: Versions are managed in the InFormALin parent

**Comment by Gram21** [comment]
Merging for now to the branch and working directly on it.

**Pull request merged**